src/RED/components/Power&BMS.tsx

```diff
@@ -213,104 +218,45 @@ class Power extends Component<IProps, IState> {
     // list and sets all their "enabled" states to true. Same if allMotorToggle() is passed false except it
     // sets all the states to false. Then the state changes all get thrown to the actual setState() and the
     // command is packed and sent.
-    if (button) {
-      for (let i = 0; i < MOTORS.length; i++) {
-        boardTelemetry = {
-          ...boardTelemetry,
-          [MOTORS[i]]: {
-            ...boardTelemetry[MOTORS[i]],
-            enabled: true,
-          },
-        }
-      }
-    } else {
-      for (let i = 0; i < MOTORS.length; i++) {
-        boardTelemetry = {
-          ...boardTelemetry,
-          [MOTORS[i]]: {
-            ...boardTelemetry[MOTORS[i]],
-            enabled: false,
-          },
-        }
-      }
-    }
+    Object.keys(boardTelemetry.MotorBusEnable).forEach((motor: string) => {
+      boardTelemetry.MotorBusEnable[motor].enabled = button ? true : false
+    })
```

Comment:
This is so much better than it used to be! The ternary is unnecessary though.
```suggestion
    Object.keys(boardTelemetry.MotorBusEnable).forEach((motor: string) => {
      boardTelemetry.MotorBusEnable[motor].enabled = button
    })
```

Reply:
fixed